src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -137,24 +244,72 @@ void containsValuesWithInlineExecutorService() {
                     }
                 ),
                 new HasValues<>("txt 1", "txt 2", "txt 3")
-            ).affirm()
-        );
+            ).affirm();
+        } finally {
+            extor.shutdownNow();
+        }
+    }
+
+    /**
+     * Execute 1 task within executor service and ensure that we'll get the
+     *  expected exception type.
+     */
+    @Test
+    void failsDueToExceptionNoTimeout() {
+        final ExecutorService extor = Executors.newSingleThreadExecutor();
+        try {
+            new Assertion<>(
+                "Must rethrow error",
+                () -> new LengthOf(
+                    new Threads<String>(
+                        extor,
+                        () -> {
+                            this.sleep();
+                            return "txt 1";
+                        },
+                        () -> {
+                            throw new IllegalStateException(
+                                "Something went wrong"
+                            );
+                        }
+                    )
+                ).value(),
+                new Throws<>(
+                    // @checkstyle LineLength (1 line)
+                    "java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Something went wrong",
+                    UncheckedIOException.class
+                )
+            ).affirm();
+        } finally {
+            extor.shutdownNow();
+        }
     }
 
     /**
-     * Repeat the test several times.
-     * @param test The test to execute.
+     * Execute the tasks concurrently using {@link Threads} when
+     *  {@link ExecutorService} was initiated by {@link Threads} itself.
      */
-    private void repeat(final Proc<?> test) {
-        new UncheckedFunc<>(
-            new Repeated<>(
-                arg -> {
-                    test.exec(null);
-                    return null;
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsValuesWithInlineExecutorServiceNoTimeout() {
+        new Assertion<>(
+            "Must contain results from the callables",
```

Comment:
@victornoel perhaps we could write : "Must contain results from the callables when using inline executor without timeout" to make a difference with test `containsResults`